libs/execution_context/ebpf_link.c

```diff
@@ -325,6 +335,16 @@ ebpf_link_attach_program(_Inout_ ebpf_link_t* link, _Inout_ ebpf_program_t* prog
 
     ebpf_assert(link->program == NULL);
 
+    link->client_data_v2.flags = ebpf_program_get_flags(program);
+    // If any of the flags are set, use the V2 data structure.
+    if (link->client_data_v2.flags != 0) {
+        link->client_characteristics.ClientRegistrationInstance.NpiSpecificCharacteristics = &link->client_data_v2;
+    }
+    // If no flags are set, use the V1 data structure.
```

Comment:
Why is it based on flags instead of, say, using a union field so the pointer is the same?
And why do you need to actually store the v1 data anyway, since it can be derived from the v2 data.
I.e., why not only store v2?

Reply:
Switched to only use the new type.